.travis.yml

```diff
@@ -8,6 +8,21 @@ jobs:
     - docker build --target test  --tag todo-app:test .
     - docker run -e SECRET_KEY -e API_KEY -e API_TOKEN todo-app:test tests_e2e
     - docker run --env-file ./.env.test todo-app:test tests
+
+  - stage: docker build and publish
```

Comment:
Both of these new stages will run on every build - that means that every pushed commit/PR will kick off a deployment which is rarely what we want, even for a staging environment. Instead, we should ensure they only run on specified branches (and not for PRs) - this can be done conditionally on these stages, but switching to [a script deploy](https://docs.travis-ci.com/user/deployment/script/) will make that easy (and save us having to worry about PRs ourselves). We could take advantage of the before_deploy step ([mentioned in the docs here](https://docs.travis-ci.com/user/job-lifecycle/#deploying-your-code)) to do our docker build/publish